monkestation/code/modules/library/skill_learning/job_skillchips/shaft_miner.dm

```diff
@@ -0,0 +1,186 @@
+
+	//Items: Misc
+	/obj/item/toy/plush/moth/ = list("Bug!", "Found a bug!", "There's a bug here!", "Moth!", "Moth here!", "🐛"),
+
+	//Machines
+	/obj/machinery/mineral/ore_redemption/ = list("ORM! Drop off your rocks and stones!", "It's a mystery to me how all these minerals fit inside!", "Making a deposit!", "Molly!", "Bloody scientists never upgrade this thing..."),
+	/obj/machinery/computer/shuttle/ = list("Shuttle console!", "Away we go!"),
+
+	//Static Objects: Icemoon
+	/obj/structure/flora/ash/chilly/ = list("Got some fruit here!", "Ice pepper plant!", "Cold as ice!", "Frosty peppers!", "Time for some chilly chili!"),
+
+	//Static Objects: Oshan
+	//None yet, but I am leaving it open.
+
+	//Static Objects: Lavaland & Other
+	/obj/structure/flora/rock/ = list("Rock! Or stone?", "Rock!", "Pioneers used to ride these babies for miles."),
+	/obj/structure/geyser/ = list("Chemical geyser!", "Geyser here! Get the chemists!", "Found a geyser!", "Got some mystery chems here!", "Geyser! Probably tastes better than the chef's cooking..."),
+	/obj/structure/flora/ash/cacti/ = list("Cacti here!", "Don't step on this one!", "Yeast Cones?", "Cactus right here!"),
+	/obj/structure/flora/ash/fireblossom/ = list("Fireblossom here!", "Got a fireblossom plant!", "Fireblossom, for when you need to glow!"),
+	/obj/structure/flora/ash/ = list("Mushroom!"), //You just know I had to do this.
+	/turf/closed/mineral/ = list("Dig through here.", "Minerals inside?", "We need to dig here.", "Drill here?")
+))
+
+/obj/item/skillchip/drg_callout
+	name = "D.R.G.R.A.S Skillchip" //Deep Rock Galactic Reactive Alert System (Or ROCK AND STONE)
+	desc = "Smells faintly of alcohol and has an odd coffee stain on it."
+	custom_price = PAYCHECK_CREW
+	complexity = 0
+	skill_name = "Miner Communication"
+	skill_description = "Understand the skills required to rapidly recognize and call out objects you've pointed at to teammates."
+	skill_icon = "bullhorn"
+	cooldown = 5 SECONDS //Honestly, this should be easy to turn off at any time if you don't want it anymore.
+	activate_message = "<span class='notice'>You suddenly understand the need to shout about things you point at.</span>"
+	deactivate_message = "<span class='notice'>You no longer understand why you were yelling so much.</span>"
+	//5-10 second delay for radio messages
+	COOLDOWN_DECLARE(radio_cooldown)
+	//1 second delay for regular point shouts
+	COOLDOWN_DECLARE(shout_cooldown)
+
+
+/obj/item/skillchip/drg_callout/on_activate(mob/living/carbon/user, silent)
+	. = ..()
+	RegisterSignal(user, COMSIG_MOB_POINTED, PROC_REF(point_handler))
+
+/obj/item/skillchip/drg_callout/proc/point_handler(mob/pointing_mob, atom/pointed_at)
+	SIGNAL_HANDLER
+
+	if(!COOLDOWN_FINISHED(src, shout_cooldown))
+		return
+
+	var/type = is_path_in_list_return_path(pointed_at.type, GLOB.miner_callouts)
+	if(!type)
+		return
+	var/list/callouts = GLOB.miner_callouts[type]
+	if(!length(callouts))
```

Comment:
```suggestion
	var/list/callouts = GLOB.miner_callouts[pointed_at.type]
	if(!callouts || !length(callouts))
```
BYOND can just directly check for key value within a list and will give `null` if there is none, so the `type` stuff can be skipped.

Reply:
That method does not work for subtypes. Doing it in this way has allowed for subtypes to be caught, rather than tripling the length of the list to catch every possible subtype.
This was actually an issue found with the original /tg/ PR.